kotlin/find-route/src/main/java/com/esri/arcgisruntime/sample/findroute/MainActivity.kt

```diff
@@ -0,0 +1,321 @@
+      bottomSheetBehavior.peekHeight = bottomSheet.header.height
+      mainContainerParams.bottomMargin = bottomSheet.header.height
+      solveRoute()
+    }
+  }
+
+  /**
+   * Solves the route using a Route Task, populates the navigation drawer with the directions,
+   * and displays a graphic of the route on the map.
+   */
+  private fun solveRoute() {
+    // create a route task instance
+    val routeTask = RouteTask(this, getString(R.string.routing_service))
+    // create an alert dialog for showing load progress
+    val progressDialog = createProgressDialog(routeTask)
+    progressDialog.show()
+
+    val listenableFuture = routeTask.createDefaultParametersAsync()
+    listenableFuture.addDoneListener {
+      try {
+        if (listenableFuture.isDone) {
+          val routeParams = listenableFuture.get()
+          // create stops
+          val stops = arrayListOf(
+            Stop(Point(-117.15083257944445, 32.741123367963446, SpatialReferences.getWgs84())),
+            Stop(Point(-117.15557279683529, 32.703360305883045, SpatialReferences.getWgs84()))
+          )
+          routeParams.apply {
+            setStops(stops)
+            // set return directions as true to return turn-by-turn directions in the route's directionManeuvers
+            isReturnDirections = true
+          }
+
+          // solve the route
+          val result = routeTask.solveRouteAsync(routeParams).get()
+          val route = result.routes[0] as Route
+          // create a simple line symbol for the route
+          val routeSymbol = SimpleLineSymbol(SimpleLineSymbol.Style.SOLID, Color.BLUE, 5f)
+          // create a graphic for the route
+          Graphic(route.routeGeometry, routeSymbol).also {
+            // add the graphic to the map
+            graphicsOverlay.graphics.add(it)
+          }
+
+          // get directions
+          // NOTE: to get turn-by-turn directions the route parameters
+          //  must have the isReturnDirections parameter set to true.
+          val directions = route.directionManeuvers
+          val directionsArray = Array<String>(directions.size) { i ->
+            directions[i].directionText
+          }
+
+          if (progressDialog.isShowing) {
+            progressDialog.dismiss()
+          }
+
+          // expand or collapse the bottom sheet when the header is clicked
+          bottomSheet.header.setOnClickListener {
+            bottomSheetBehavior.state = when (bottomSheetBehavior.state) {
+              BottomSheetBehavior.STATE_COLLAPSED -> BottomSheetBehavior.STATE_EXPANDED
+              else -> BottomSheetBehavior.STATE_COLLAPSED
+            }
+          }
+
+          bottomSheet.directionsListView.apply {
+            // Set the adapter for the list view
+            adapter = ArrayAdapter(
+              this@MainActivity,
+              android.R.layout.simple_list_item_1,
+              directionsArray
+            )
```

Comment:
I think we should hide this in a method to keep the focus on our API code. `bottomSheetHandler` or something like that if it's possible to extract it?

Reply:
see above